modules/common-transformer-stream/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/stream/common/Processing.scala

```diff
@@ -52,26 +53,27 @@ object Processing {
   type TransformationResults[C] = (List[TransformationResult], C)
   type SerializationResults[C] = (List[(SinkPath, Transformed.Data)], State[C])
 
-  def run[F[_]: ConcurrentEffect: ContextShift: Clock: Timer: Parallel, C: Checkpointer[F, *]](
+  def run[F[_]: Async: Parallel, C: Checkpointer[F, *]](
     resources: Resources[F, C],
     config: Config,
     processor: Processor
   ): Stream[F, Unit] = {
     val source = resources.inputStream.read.map(m => (parseEvent(m.content, processor), resources.checkpointer(m)))
-    runFromSource(source, resources, config, processor)
+    runFromSource(source, resources, config, processor, Clock[F])
   }
 
-  def runFromSource[F[_]: ConcurrentEffect: ContextShift: Clock: Timer: Parallel, C: Checkpointer[F, *]](
+  def runFromSource[F[_]: Async, C: Checkpointer[F, *]](
     source: Stream[F, ParsedC[C]],
     resources: Resources[F, C],
     config: Config,
-    processor: Processor
+    processor: Processor,
+    clock: Clock[F]
   ): Stream[F, Unit] = {
     val transformer: Transformer[F] = config.formats match {
       case f: TransformerConfig.Formats.Shred =>
-        Transformer.ShredTransformer(resources.igluResolver, resources.propertiesCache, f, resources.atomicLengths, processor)
+        Transformer.ShredTransformer(resources.igluResolver, resources.propertiesCache, f, resources.atomicLengths, processor, clock)
```

Comment:
Is there a reason you chose to pass the clock explicitly instead of implicitly?  Like....

```scala
 def runFromSource[F[_]: Async: Clock, C: Checkpointer[F, *]](
    source: Stream[F, ParsedC[C]],
    resources: Resources[F, C],
    config: Config,
    processor: Processor
  ): Stream[F, Unit] =
```

Reply:
Unlike CE2, Clock is part of the typeclass hierarchy in CE3 where Async extends Sync extends Clock and we need to mock Clock instance in our tests, which led me to pass Clock instances explicitly

Reply:
otherwise Clock instance is resolved from Async/Sync instance which doesn't fit our tests

I'm also in favor of consistency and moved explicit clock instance around even when it isn't necessary (when there is no Async/Sync constraint e.g. ShredTransformer)

Reply:
In CE3, TestControl is introduced to mock time but it isn't a drop-in replacement for TestContext, and I picked passing clock explicitly over re-writing some tests

Reply:
Ah I see, thanks for explaining.  If we need to pass around a `Async` and a `Clock` together -- even though Async extends Clock -- then it probably means we're doing something wrong in our tests.  I guess the expected way to handle this in tests is by following [these instructions](https://github.com/snowplow/iglu-scala-client/blob/2.2.1/modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/RegistryLookup.scala#L91) for mocking time with a TestControl.